File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        no_take, take = 0, cost[0]
        i = 1
        while i < len(cost):
            dummy = take
            take = min(no_take, take) + cost[i]
            no_take = dummy
            i += 1
        return min(take, no_take)

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    let [take, noTake] = [cost[0], 0];
    
    for (let i = 1; i < cost.length; i++){
        let temp = take;
        take = Math.min(take, noTake) + cost[i];
        noTake = temp;
    }
    
    return Math.min(take, noTake);
};